src/Impl/MessageStatus.php

```diff
@@ -46,10 +46,10 @@ public function jsonSerialize(): array
 
     public static function fromJson(StdClass $json): MessageStatus
     {
-        $dataValidator = DataValidator::for($json);
+        $propertyRetriever = PropertyRetriever::for($json);
         return new MessageStatus(
-            SuccessResult::fromJson($dataValidator->retrieveNullable('success')),
-            FailureResponse::fromJson($dataValidator->retrieveNullable('failure'))
```

Comment:
This bugged me - there was no validation, just retrieving an object's property.

Minor refactor